xcm/xcm-executor/src/traits/conversion.rs

```diff
@@ -203,3 +204,24 @@ impl<O> ConvertOrigin<O> for Tuple {
 		Err(origin)
 	}
 }
+
+/// Defines how a call is dispatched with given origin.
+/// Allows to customize call dispatch, such as adapting the origin based on the call
+/// or modifying the call.
+pub trait CallDispatcher<Call: Dispatchable> {
+	fn dispatch(
+		call: Call,
+		origin: Call::Origin,
+	) -> Result<Call::PostInfo, DispatchErrorWithPostInfo<Call::PostInfo>>;
+}
+
+/// Just call `call.dispatch(origin)`.
+pub struct JustDispatch;
+impl<Call: Dispatchable> CallDispatcher<Call> for JustDispatch {
+	fn dispatch(
+		call: Call,
+		origin: Call::Origin,
+	) -> Result<Call::PostInfo, DispatchErrorWithPostInfo<Call::PostInfo>> {
+		call.dispatch(origin)
+	}
+}
```

Comment:
```suggestion
impl<Call: Dispatchable> CallDispatcher<Call> for Call {
	fn dispatch(
		call: Call,
		origin: Call::Origin,
	) -> Result<Call::PostInfo, DispatchErrorWithPostInfo<Call::PostInfo>> {
		call.dispatch(origin)
	}
}
```

Then we don't need any extra type :P